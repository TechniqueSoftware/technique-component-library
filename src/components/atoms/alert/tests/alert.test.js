import React from 'react'
import { render, screen } from '@testing-library/react'
import Alert from '../index'

const propVariations = {
  error: {
    code: 'FORM_SUBMISSION_ERROR',
    'First Name': 'This field is required',
    'Last Name': 'This field is required',
    Email: 'Please provide an email or phone number',
    Phone: 'Please provide an email or phone number'
  },
  success: {
    code: 'SUCCESS',
    firstName: 'John',
    lastName: 'Doe',
    email: 'jdoe0@aol.com'
  },
  duplicate: {
    code: 'DUPLICATE_USER',
    user: {
      userId: 172194229,
      firstName: 'Geoff',
      lastName: 'Peel'
    }
  }
}

describe('<Alert>', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Alert alertData={propVariations.success} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders a success alert', () => {
    render(<Alert alertData={propVariations.success} />)
    expect(screen.getByRole('alert')).toBeInTheDocument()
  })

  it('renders an error alert', () => {
    render(<Alert alertData={propVariations.error} />)
    expect(screen.getByRole('alert')).toBeInTheDocument()
    expect(screen.getByRole('alert')).toHaveClass('alert alert-danger')
  })

  it('renders a duplicate user alert', () => {
    render(<Alert alertData={propVariations.duplicate} />)
    expect(screen.getByRole('alert')).toBeInTheDocument()
    expect(screen.getByRole('alert')).toHaveClass('alert alert-warning')
  })
})
