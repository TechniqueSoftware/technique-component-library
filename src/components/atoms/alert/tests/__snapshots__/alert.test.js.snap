// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="fade alert alert-success show"
    role="alert"
  >
    <p>
      <strong>
        <a
          ;=""
          class="alert-link"
          false;"=""
          href="#"
          onclick="delegate(undefined, '/action/Dashboard')"
          return=""
          role="button"
          tabindex="0"
        >
          John Doe
        </a>
         has been added!
      </strong>
    </p>
  </div>
</DocumentFragment>
`;
